lib/jwt/signature.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+require 'openssl'
+
+module JWT
+  module Signature
+    extend self
```

Comment:
Using `module_functions` is the preferred coding style.

You're alway able to define a function as private by using this syntax:

```ruby
def my_fun(my_param) do
  puts my_param
end
private :my_fun
```